oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/NodeDocumentRevisionCleaner.java

```diff
@@ -0,0 +1,270 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.plugins.document;
+
+import org.apache.jackrabbit.oak.plugins.document.util.Utils;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+public class NodeDocumentRevisionCleaner {
+
+    private final DocumentStore documentStore;
+    private final DocumentNodeStore documentNodeStore;
+    private final NodeDocument workingDocument;
+    private final RevisionPropertiesClassifier revisionClassifier;
+    private final RevisionCleanerUtility revisionCleaner;
+
+    /**
+     * Constructor for NodeDocumentRevisionCleaner.
+     * @param documentStore The DocumentStore instance. Must be writable to perform cleanup.
+     * @param documentNodeStore The DocumentNodeStore instance.
+     * @param workingDocument The document to clean up.
+     */
+    public NodeDocumentRevisionCleaner(DocumentStore documentStore, DocumentNodeStore documentNodeStore,
+                                       NodeDocument workingDocument) {
+        this.workingDocument = workingDocument;
+        this.documentStore = documentStore;
+        this.documentNodeStore = documentNodeStore;
+
+        revisionClassifier = new RevisionPropertiesClassifier(workingDocument);
+        revisionCleaner = new RevisionCleanerUtility(revisionClassifier);
+    }
+
+    /**
+     * Collects cleanable old revisions for the given document.
+     */
+    public void collectOldRevisions(UpdateOp op) {
+        revisionClassifier.classifyRevisionsAndProperties();
+        revisionCleaner.preserveRevisionsNewerThanThreshold(24, ChronoUnit.HOURS);
+        revisionCleaner.preserveLastRevisionForEachProperty();
+        revisionCleaner.preserveRevisionsReferencedByCheckpoints();
+        revisionCleaner.removeCandidatesInList();
+
+        /*for (Map.Entry<Integer, TreeSet<Revision>> entry : revisionCleaner.candidateRevisionsToClean.entrySet()) {
+            for (Revision revision : entry.getValue()) {
+                System.out.println("Removing revision " + revision);
+                TreeSet<String> properties = revisionClassifier.propertiesModifiedByRevision.get(revision);
+                System.out.println("Properties modified by this revision: " + properties);
+                if (properties != null) {
+                    for (String property : properties) {
+                        op.removeMapEntry(property, revision);
+                    }
+                }
+
+                op.removeMapEntry("_revisions", revision);
+            }
+        }*/
+    }
+
+    protected class RevisionPropertiesClassifier {
```

Comment:
Is there a reason this is not `private` ?

Reply:
Left over from initial testing. I have made it private.